File: cpp/Problem217.cpp
/*
 * Problem: 217. Contains Duplicate
 * URL: https://leetcode.com/problems/contains-duplicate/
 * Difficulty: Easy
 * Tags: Array, Set
 */

#include <vector>
#include <unordered_set>

using namespace std;

class Solution {
public:
	bool containsDuplicate(vector<int>& nums) {
		unordered_set<int> seen;
		for (int num : nums) {
			if (seen.count(num) > 0) {
				return true;
			}
			seen.insert(num);
		}
		return false;
	}
};

File: python/Problem217.py
# Problem: 217. Contains Duplicate
# URL: https://leetcode.com/problems/contains-duplicate/
# Difficulty: Easy
# Tags: Array, Set

from typing import List

class Solution:
  def containsDuplicate(self, nums: List[int]) -> bool:
    seen: set[int] = set()
    for num in nums:
      if num in seen:
        return True
      seen.add(num)
    return False